test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -262,19 +316,25 @@ static function (SplFileInfo $file) use ($zip) {
 
     private function createArtifact() : void
     {
+        $this->createGenericArtifact(
+            'test-package',
+            '2.0.0',
+            [
+                'name' => 'test/package',
+            ]
+        );
+    }
+
+    /**
+     * @param mixed[] $composerJsonContent
+     */
+    private function createGenericArtifact(string $name, string $version, array $composerJsonContent): void
```

Comment:
We could use `'path'` to force composer to perform a local (symlink) installation or such, instead of having to provide actual zip files.

Alternatively, the packages could be of `'type' => 'metapackage'`, which doesn't lead to any file installation (https://github.com/Roave/SecurityAdvisories/blob/666ba252853924887ac57dc9f66e6b6af78d5a76/composer.json#L3)

Reply:
I think, I can try `path`. Though I still don't get the idea what for

Reply:
I'm trying to remove even further components from the patch, that's all. Maintaining these tests, especially if many things are being moved around, is going to be a challenge in the years to come

Reply:
Sorry for long delay, got back to this issue. I stumbled upon a question, should I rewrite all zips (even original, which presents before my changes) or only new packages I test (so there will be both zip for package versions artifact and path for other packages)?

Reply:
No worries, I'm also not actively watching this issue ATM: my suggestion is to get to a point where no sources are needed at all, and only the `composer.lock` and `installed.json` are produced along with the generated versions.